app/src/main/java/com/zeugmasolutions/localeexample/MainActivity.kt

```diff
@@ -15,16 +16,46 @@ class MainActivity : BaseActivity() {
 
         setTitle(R.string.main_activity_title)
 
-        toTRButton.setOnClickListener { updateLocale(Locales.Turkish) }
-        toENButton.setOnClickListener { updateLocale(Locale.ENGLISH) }
-        toCNButton.setOnClickListener { updateLocale(Locale.CHINA) }
-        toURButton.setOnClickListener { updateLocale(Locales.Urdu) }
+        language_picker.setOnCheckedChangeListener { _, checkedId ->
+            when(checkedId) {
+                R.id.language_default -> clearLocaleSelection()
```

Comment:
If you change this to updateLocale(LocaleHelper.systemLocale) you can run this example app with the existing library to replicate the issues.